### PR TITLE
pass list of certificates to Orca rather than single certificateArn

### PR DESCRIPTION
**Pull request by robfletcher**
This is an alternate implementation to #1736 where I'm not changing the ALB spec model, just amending how we pass the cert to Orca.

### CODE REVIEW

File: keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandlerTests.kt [hunk 1]
```diff
@@ -92,13 +92,14 @@ internal class ApplicationLoadBalancerHandlerTests : JUnit5Minutests {
     |    - us-east-1d
     |listeners:
     | - port: 80
-    |   protocol: HTTP
+    |   protocol: HTTPS
+    |   certificateArn: this-is-my-certificate
```
Comment (emjburns) [inline keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandlerTests.kt:96]: I think i like this change better. there will be no change in the config, just a change on how we call orca so things work?

Comment (robfletcher) [reply keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandlerTests.kt:96]: Correct

Comment (robfletcher) [reply keel-ec2-plugin/src/test/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandlerTests.kt:96]: I would still like to figure out how to better specify a cert (if possible), but this should get us unblocked

**Comment by lorin** [comment]
I agree with @emjburns, I prefer this approach to #1736. This one:

* has a simpler delivery config format
* the change is much less intrusive

The only downside I see compared to #1736 is that this approach doesn't support multiple certificates. However, I'd prefer to wait until we actually get a request for that functionality, since YAGNI.

File: keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt [hunk 2]
```diff
@@ -294,7 +294,20 @@ class ApplicationLoadBalancerHandler(
...
           "securityGroups" to dependencies.securityGroupNames,
-          "listeners" to listeners,
+          "listeners" to listeners.map {
+            mapOf(
+              "port" to it.port,
+              "protocol" to it.protocol,
+              "rules" to it.rules,
+              "defaultActions" to it.defaultActions,
+            ).run {
+              if (it.certificateArn == null) {
```
Comment (lorin) [inline keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ApplicationLoadBalancerHandler.kt:304]: I love that `if` statements are expressions in Kotlin.

**Review by lorin — APPROVED** [review]
LGTM

**Review by emjburns — APPROVED** [review]

**Pull request merged**